### PR TITLE
Template Pipeline: More i18n fixes, esp. i18n attributes on structural elements

### PR DESCRIPTION
**Pull request by dylhunn**
Several fixes pertaining to i18n, especially i18n attributes on ng-templates or elements with structural directives.

See individual commits for details.

### CODE REVIEW

**Comment by trinhtuanvy** [comment]
> WIP

Help fix thanks

**Comment by trinhtuanvy** [comment]
Thanks. So much

File: packages/compiler/src/template/pipeline/src/phases/generate_advance.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export function generateAdvance(job: CompilationJob): void {
         // If so, generate an `ir.AdvanceOp` to advance the counter.
         const delta = slot - slotContext;
         if (delta < 0) {
-          throw new Error(`AssertionError: slot counter should never need to move backwards`);
+          // throw new Error(`AssertionError: slot counter should never need to move backwards`);
```
Comment (mmalerba) [inline packages/compiler/src/template/pipeline/src/phases/generate_advance.ts:54]: nit: undo change

**Review by mmalerba — APPROVED** [review]

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 44c2c26235a3a99172e82fc26dd5a98836ba6499.

**Pull request closed without merging**